mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -256,6 +239,96 @@
         return ResponseEntity.ok(new Resource<>(applicationDeploymentInformation));
     }
 
+    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}")
+    public ResponseEntity<Resources<Resource<MicoApplication>>> getApplicationsByShortName(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) {
+        List<MicoApplication> micoApplicationList = applicationRepository.findByShortName(shortName);
+
+        List<Resource<MicoApplication>> applicationResourceList = getApplicationResourceList(micoApplicationList);
+
+        return ResponseEntity.ok(
+            new Resources<>(applicationResourceList,
+                linkTo(methodOn(ApplicationController.class).getApplicationsByShortName(shortName)).withSelfRel()));
+    }
+
+    @PatchMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/" + PATH_SERVICES)
+    public ResponseEntity<Void> addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String applicationShortName,
+                                                        @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                        @RequestBody MicoService service) {
+        Optional<MicoApplication> applicationOptional = applicationRepository.findByShortNameAndVersion(applicationShortName, applicationVersion);
+        if (!applicationOptional.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND,
+                "Application '" + applicationShortName + "' '" + applicationVersion + "' does not exist!");
+        }
+        validateProvidedService(service);
+
+        MicoApplication application = applicationOptional.get();
+        if (!application.getServices().contains(service)) {
+            log.info("Add service '" + service.getShortName() + "' '" + service.getVersion() +
+                "' to application '" + applicationShortName + "' '" + applicationVersion + "'.");
+            application.getServices().add(service);
+            applicationRepository.save(application);
+        } else {
```

Comment:
Is the error in the else case not a potential 4XX error? Is there intentionally no return to make the request idempotent? Also, you could consider using the warning log level, because the UI should not allow adding a service to an application if it already exists. This state could indicate an error.

Reply:
I would be in favor of idempotent bahaviour here

Reply:
If we created a complete new resource, I would throw a 409 Conflict error. However here we just want to create a link between two existing resources. I think it does not make sense to throw an error if this link is already there.